flang/lib/Semantics/check-call.cpp

```diff
@@ -1033,6 +1033,14 @@ static void CheckExplicitDataArg(const characteristics::DummyDataObject &dummy,
               *actualDataAttr == common::CUDADataAttr::Managed)) {
         actualDataAttr = common::CUDADataAttr::Device;
       }
+      // For device procedures, treat non-VALUE arguments without explicit CUDA
```

Comment:
I'm confused with the non-VALUE here? We want actual with VALUE attribute to be ok with DEVICE dummy right?